qcodes/dataset/data_export.py

```diff
@@ -83,17 +84,17 @@ def get_data_by_id(run_id: int) -> List:
         dep_data_dict_index = None
 
         for param_name, data in data_dict.items():
-            data_dict = {}
+            my_data_dict: Dict[str, Union[str, np.ndarray]] = {}
```

Comment:
If we would allow a dependency on typing extensions this could be made a Typed dict which would avoid a lot of the casts needed when taking data out of this dict.

Reply:
`allow a dependency on typing extensions` - shall we "wait for" python 3.8 for that?

Reply:
Sure I fail to see why that matters but I don't really feel strongly about it